src/widgets/search-box/search-box.js

```diff
@@ -73,81 +43,82 @@ const renderer = ({
     // - use back button
     // - input query is empty (because <input> autocomplete = off)
     window.addEventListener('pageshow', () => {
-      input.value = queryFromInput;
+      input.value = query;
     });
 
     // Update value when query change outside of the input
     onHistoryChange(fullState => {
       input.value = fullState.query || '';
     });
 
-    if (autofocus === true || (autofocus === 'auto' && queryFromInput === '')) {
+    if (autofocus === true) {
       input.focus();
-      input.setSelectionRange(queryFromInput.length, queryFromInput.length);
+      input.setSelectionRange(query.length, query.length);
     }
 
-    // search on enter
-    if (searchOnEnterKeyPressOnly) {
-      addListener(input, INPUT_EVENT, e => {
+    const form = input.parentElement;
+
+    if (searchAsYouType) {
+      addListener(input, 'input', getInputValueAndCall(refine));
+      addListener(form, 'submit', e => {
+        e.preventDefault();
+        input.blur();
+      });
+    } else {
+      addListener(input, 'input', e => {
         refine(getValue(e), false);
       });
-      addListener(input, 'keyup', e => {
-        if (e.keyCode === KEY_ENTER) refine(getValue(e));
+      addListener(form, 'submit', e => {
+        refine(input.value);
+        e.preventDefault();
+        input.blur();
       });
-    } else {
-      addListener(input, INPUT_EVENT, getInputValueAndCall(refine));
-
-      // handle IE8 weirdness where BACKSPACE key will not trigger an input change..
-      // can be removed as soon as we remove support for it
-      if (INPUT_EVENT === 'propertychange' || window.attachEvent) {
-        addListener(
-          input,
-          'keyup',
-          ifKey(KEY_SUPPRESS, getInputValueAndCall(refine))
-        );
-      }
     }
-  } else {
-    renderAfterInit({
-      containerNode,
-      query,
-      loadingIndicator,
-      isSearchStalled,
-    });
   }
 
-  if (reset) {
-    const resetBtnSelector = `.${cx(bem('reset-wrapper'))}`;
-    // hide reset button when there is no query
-    const resetButton =
-      containerNode.tagName === 'INPUT'
-        ? containerNode.parentNode.querySelector(resetBtnSelector)
-        : containerNode.querySelector(resetBtnSelector);
-    resetButton.style.display = query && query.trim() ? 'block' : 'none';
-  }
+  renderAfterInit({
```

Comment:
With the previous implement we trigger this function only on render but now it also trigger on init, not sure about the implications. But we almost never render on the init lifecycle to avoid a flash with the empty state.